engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCache.scala

```diff
@@ -98,8 +98,54 @@ class CallCache(database: CallCachingSqlDatabase) {
   def invalidate(callCachingEntryId: CallCachingEntryId)(implicit ec: ExecutionContext) = {
     database.invalidateCall(callCachingEntryId.id)
   }
+
+  def callCacheDiff(callA: CallCacheDiffQueryCall, callB: CallCacheDiffQueryCall)(implicit ec: ExecutionContext) = {
+    database.diffCallCacheHashes(
+      callA.workflowId, callA.callFqn, callA.jobIndex.fromIndex,
+      callB.workflowId, callB.callFqn, callB.jobIndex.fromIndex
+    )
+  }
 }
 
 object CallCache {
-  case class CallCacheHashBundle(workflowId: WorkflowId, callCacheHashes: CallCacheHashes, response: JobSucceededResponse)
+  object CallCacheHashBundle {
+    def apply(workflowId: WorkflowId, callCacheHashes: CallCacheHashes, jobSucceededResponse: JobSucceededResponse) = {
+      new CallCacheHashBundle(
+        workflowId,
+        callCacheHashes,
+        jobSucceededResponse.jobKey.call.fullyQualifiedName,
+        jobSucceededResponse.jobKey.index,
+        Option(jobSucceededResponse.jobKey.attempt),
+        jobSucceededResponse.returnCode,
+        true,
+        jobSucceededResponse.jobOutputs,
+        jobSucceededResponse.jobDetritusFiles
+      )
+    }
+
+    def apply(workflowId: WorkflowId, callCacheHashes: CallCacheHashes, jobFailedNonRetryableResponse: JobFailedNonRetryableResponse) = {
+      new CallCacheHashBundle(
+        workflowId,
+        callCacheHashes,
+        jobFailedNonRetryableResponse.jobKey.scope.fullyQualifiedName,
+        jobFailedNonRetryableResponse.jobKey.index,
+        Option(jobFailedNonRetryableResponse.jobKey.attempt),
+        None,
+        false,
+        Map.empty[LocallyQualifiedName, JobOutput],
+        None
+      )
+    }
+  }
+  case class CallCacheHashBundle private (
```

Comment:
Oh I see, the reason for the private was to try to make it hard to mess up the value of "allowedResultReuse" when creating a `CallCacheHashBundle`, as this is gonna be even more important now as we really don't want failed jobs to be cacheable. It at least adds *some* safety I guess.